src/inspector_agent.cc

```diff
@@ -625,6 +605,37 @@ void Agent::RegisterAsyncHook(Isolate* isolate,
                               v8::Local<v8::Function> disable_function) {
   enable_async_hook_function_.Reset(isolate, enable_function);
   disable_async_hook_function_.Reset(isolate, disable_function);
+  if (pending_enable_async_hook_) {
+    pending_enable_async_hook_ = false;
+    EnableAsyncHook(isolate);
+  } else if (pending_disable_async_hook_) {
+    pending_disable_async_hook_ = false;
+    DisableAsyncHook(isolate);
+  }
+}
+
+void Agent::EnableAsyncHook(Isolate* isolate) {
+  CHECK(!enable_async_hook_function_.IsEmpty());
+  Local<Function> enable_fn = enable_async_hook_function_.Get(isolate);
+  auto context = parent_env_->context();
+  auto result = enable_fn->Call(context, Undefined(isolate), 0, nullptr);
+  if (result.IsEmpty()) {
+    FatalError(
+      "node::inspector::Agent::EnableAsyncHook",
+      "Cannot enable Inspector's AsyncHook, please report this.");
+  }
+}
+
+void Agent::DisableAsyncHook(Isolate* isolate) {
+  CHECK(!disable_async_hook_function_.IsEmpty());
+  Local<Function> disable_fn = disable_async_hook_function_.Get(isolate);
+  auto context = parent_env_->context();
+  auto result = disable_fn->Call(context, Undefined(isolate), 0, nullptr);
+  if (result.IsEmpty()) {
+    FatalError(
+      "node::inspector::Agent::DisableAsyncHook",
+      "Cannot disable Inspector's AsyncHook, please report this.");
+  }
```

Comment:
These two methods could be dedup'd, e.g., by passing the `Persistent<Function>` to use as a parameter (and the name in the error message as a string.)

Reply:
Done.